ykcompile/src/lib.rs

```diff
@@ -252,6 +258,8 @@ impl TraceCompiler {
             }
             Statement::Return => self.c_return()?,
             Statement::Call(op, args, dest) => self.c_call(op, args, dest)?,
+            Statement::StorageLive(_) => {},
+            Statement::StorageDead(l) => self.free_register(l),
```

Comment:
What about

```rust
_1: u8;

bb0: {
    StorageLive(_1);
    _1 = 42u8;
    Goto bb2;
}

bb1: {
    StorageDead(_1);
    return;
}

bb2: {
    Use(_1);
    Goto ebb1;
}

Reply:
As I understand, the input to the trace compiler is a trace, so there are no blocks to speak of. Everything should have been inlined.

The above would be:
```
    StorageLive(_1);
    _1 = 42u8;
    Use(_1);
    Goto ebb1;
    StorageDead(_1);
    return;
```

Is that right @ptersilie ?

Reply:
/me forgot that this is a trace compiler and not a normal compiler.

Reply:
No worries! You got me thinking for a moment there!